circleci/client.go

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2019 Jobteaser
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package circleci expose function to manage environment variable in
+// CircleCI project.
+package circleci // import "github.com/jobteaser-oss/circleci-env"
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// Client contains anm `http.Client` and a URL.
+type Client struct {
+	http *http.Client
+	url  *url.URL
+}
+
+// NewClient returns a pointer of `Client` struct with timeout and
+// authentication properly configured.
+func NewClient(token string) (*Client, error) {
+	const timeout = 5 * time.Second
+
+	dialer := net.Dialer{Timeout: timeout}
+
+	transport := http.Transport{
+		Dial:                dialer.Dial,
+		TLSHandshakeTimeout: timeout,
+	}
+
+	client := http.Client{
+		Transport: &transport,
+		Timeout:   timeout,
+	}
+
+	uri, err := url.Parse("https://circleci.com/api/v1.1")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse the CircleCI base URL")
+	}
+
+	params := url.Values{}
+	params.Add("circle-token", token)
+	uri.RawQuery = params.Encode()
+
+	return &Client{
+		http: &client,
+		url:  uri,
+	}, nil
+}
+
+// Env represent the CircleCI API response.
+type Env struct {
+	Key   string `json:"name"`
+	Value string `json:"value"`
+}
+
+// ListEnv returns four 'x' characters plus the last four ASCII characters
+// of the value, consistent with the display of environment variable values
+// in the CircleCI website.
```

Comment:
I found this hard to understand.

May I suggest

    LienEnv lists the environment variables, with names and obfuscated values. Note: obfuscation
    is consistent with what is done on the CircleCI UI : four x characters, followed by the last
    four characters of the actual value.